tuf/api/metadata.py

```diff
@@ -1057,9 +1056,32 @@ def to_dict(self) -> Dict[str, Any]:
             res_dict["path_hash_prefixes"] = self.path_hash_prefixes
         return res_dict
 
+    @staticmethod
+    def _is_target_in_pathpattern(targetpath: str, pathpattern: str) -> bool:
+        """Determines whether "targetname" matches the "pathpattern"."""
+        # We need to make sure that targetname and pathpattern are pointing to
+        # the same directory as fnmatch doesn't threat "/" as a special symbol.
+        target_parts = targetpath.split("/")
+        pattern_parts = pathpattern.split("/")
+        if len(target_parts) != len(pattern_parts):
+            return False
+
+        # Every part in the pathpattern could include a glob pattern, that's why
+        # each of the target and pathpattern parts should match.
+        for target_dir, pattern_dir in zip(target_parts, pattern_parts):
+            if not fnmatch.fnmatch(target_dir, pattern_dir):
+                return False
+
+        return True
+
     def is_delegated_path(self, target_filepath: str) -> bool:
         """Determines whether the given 'target_filepath' is in one of
-        the paths that DelegatedRole is trusted to provide"""
+        the paths that DelegatedRole is trusted to provide.
+
+        Note: we only support "/" as a separator and don't handle corner cases
+        such as leading separators in pathpattern or target_filepath. See:
+        https://theupdateframework.github.io/specification/latest/#targetpath
```

Comment:
documentation should maybe not imply that there are _other unspecified_ "corner cases" that we do not handle.

```suggestion
        target_filepath (and the DelegatedRole paths) are expected to be in 
        their canonical forms, so e.g. "a/b" instead of "a//b" . Only
        "/" is supported as target path separator. Leading separators are not
        handled as special cases (see `TUF specification on targetpath
        https://theupdateframework.github.io/specification/latest/#targetpath`_).
```
Maybe that covers most of what we want to say?

Btw, I'm not sure I got the sphinx link format correct there

Reply:
I used almost the same text as you suggested, but for the link, I had to read the sphinx docs:
https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages.
Now, the link looks good.